src/main/java/com/minecolonies/coremod/colony/buildings/BuildingGuardTower.java

```diff
@@ -503,16 +498,35 @@ public void setAssignManually(final boolean assignManually)
     }
 
     /**
+     * Possible job/AI.
+     */
+    public enum GuardJob
+    {
+        KNIGHT,
+        RANGER,
+    }
+
+    /**
+     * Possible tasks.
+     */
+    public enum Task
+    {
+        FOLLOW,
+        GUARD,
+        PATROL
+    }
+
+    /**
      * The client view for the baker building.
      */
     public static class View extends AbstractBuildingWorker.View
     {
-        public boolean assignManually = false;
-        public boolean retrieveOnLowHealth = false;
-        public boolean patrolManually = false;
-        public Task task = Task.GUARD;
-        public GuardJob job = null;
-        public BlockPos guardPos = getLocation();
+        public boolean  assignManually      = false;
+        public boolean  retrieveOnLowHealth = false;
+        public boolean  patrolManually      = false;
+        public Task     task                = Task.GUARD;
+        public GuardJob job                 = null;
+        public BlockPos guardPos            = getLocation();
 
         public ArrayList<BlockPos> patrolTargets = new ArrayList<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "patrolTargets" object should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1319)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make patrolTargets a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)